### ISSUE TITLE
Show the word and the intermediate ranking at the end of each round

### ISSUE THREAD

**Issue by Luni-4** [scribble-rs/scribble.rs#75]
At the end of each round, it would be helpful to open a new window (as the one used to choose words) to show the word and the intermediate ranking. 

This window should remain opened for a certain amount of time set by a timer.

**Comment by Bios-Marcel** [comment]
Hm, that sounds a bit overkill to be honest. While I am agreeing to a final ranking screen, one every round seems a bit too much :man_shrugging: 

Does this really bring benefit?

**Comment by Luni-4** [comment]
> Hm, that sounds a bit overkill to be honest. While I am agreeing to a final ranking screen, one every round seems a bit too much man_shrugging
> 
> Does this really bring benefit?

This feature could be placed between the end of a round and the beginning of another one. Why should it be overkill? We only print a word and a series of strings. I thought to insert a timer to close the window automatically (after a certain amount of time necessary to the players to read the ranking)

About the benefit, I think it could be pretty useful for a player to know where he is in the ranking at the end of each round such that he's willing to try harder to increase his scores in the next round.

**Comment by Bios-Marcel** [comment]
But if he wants to know his rank, he can always look on the left. The scoreboard never gets reordered either, so he's always in the same spot.

The reason I am against adding feature that provide little to no benefit, is because somebody has to write that feature and maintain it. Every time I do a change, I might break that feature, meaning I will have to test it and probably fix it. There's always a trade-off with everything you do in an application :D

**Comment by Luni-4** [comment]
> But if he wants to know his rank, he can always look on the left. The scoreboard never gets reordered either, so he's always in the same spot.

Yeah, this could be sufficient effectively, perhaps could we only show the points gained during the round into this window or would it provide little benefit the same?

> 
> The reason I am against adding feature that provide little to no benefit, is because somebody has to write that feature and maintain it. Every time I do a change, I might break that feature, meaning I will have to test it and probably fix it. There's always a trade-off with everything you do in an application :D

Sure, this is a good argument. Perhaps we could only define a backend function that notify the frontend to show the complete word at the top when a round finishes? Because printing it in the "guessing" chat it's pretty uncomfortable according to me.

**Comment by Bios-Marcel** [comment]
Well, i didn't want there to be a phase of nothing happening between the end of a turn and the start of a new turn. It would be a waste of time. Maybe we could display a dialog exactly til a new word has been chosen :man_shrugging:

**Comment by Bios-Marcel** [comment]
For the last word I mean. Not the scoreboard, I still think that's not useful.

**Comment by Luni-4** [comment]
Okay, we could only show the word. The phase should be the following ones in my opinion:
1. The previous word is guessed or the time is out
2. Show the window containing the previous word for n seconds (less than 10)
3. The next player chooses the new word to draw between the three proposed

**Comment by zouglou** [comment]
It's kind of related, I did a pull request #119 adding a screen saying who is choosing a word.
We may add to it the previous word, and scores?

**Comment by Luni-4** [comment]
@zouglou 

Seems a good idea to me, thanks! :)

**Comment by Bios-Marcel** [comment]
Like mentioned in the PR, i was working on some changes here and I think these are necessary before we can do this well. I don't wanna make any compromises with the new gameflow.
